src/tests/math/math_test_fp.c

```diff
@@ -0,0 +1,89 @@
+/**
+ * @math_test_fp
+ * @simple tests for fp functions in embox just to make sure the corresponding functions are not stubs
+ * @date 03.03.2020
+ * @author Rushil kvs
+ */
+
+#include <embox/test.h>
+#include <math.h>
+
+EMBOX_TEST_SUITE("Tests for fp functions");
+
+bool Equal(double a, double b)
+{
+	double epsilon = 0.00000000000001;
+	if ((a-b) < epsilon && -(a-b) < epsilon) return true;
+  	return false;
+}
+
+bool Equalf(float a, float b)
+{
+	float epsilon = 0.00000000001;
+	if ((a-b) < epsilon && -(a-b) < epsilon) return true;
+  	return false;
+}
+
+long double Equall(long double a, long double b){
+	long double epsilon = 0.0000000000000000000001;
+	if ((a-b) < epsilon && -(a-b) < epsilon) return true;
+  	return false;
+}
+
+TEST_CASE("Test for modf") {
+	double i,t,j;
+	i = modf(8.123456,&t);
+	j = 0.123456;
+	test_assert_true(Equal(i,j));
+	test_assert_false(Equal(i,-j));
+	test_assert_true(Equal(t,8.0));
+	test_assert_false(Equal(t,-8.0));
+}
+
+TEST_CASE("Test for fmod") {
+	double i;
+	//bool b;
+	i = fmod(8.432,3.112);
```

Comment:
Speaking about of simplifications with epsilons I just mean that you can try to change this one

```
i = fmod(8.432,3.112);
test_assert_true(Equal(i,2.208));
```

to this one:

```
i = fmod(5.0,2.0);
test_assert_true(i, 2.5);
```